File: PcrApp/app/src/main/java/com/example/vishaalprasad/pcrapp/reactant_helpers/MissingStockConcentrationException.java
package com.example.vishaalprasad.pcrapp.reactant_helpers;

/**
 * Thrown by a {@link ConcentrationReactant} if is missing a {@link StockConcentration}
 */
public class MissingStockConcentrationException extends Exception {

    private String reactantName;

    public MissingStockConcentrationException(Reactant reactant){
        this.reactantName = reactant.getClass().getSimpleName();
    }

    @Override
    public String getMessage() {
        return reactantName + " reactant is missing a stock concentration";
    }
}

File: PcrApp/app/src/main/java/com/example/vishaalprasad/pcrapp/reactant_helpers/MissingStockConcentrationException.kt
package com.example.vishaalprasad.pcrapp.reactant_helpers

/**
 * Thrown by a [ConcentrationReactant] if is missing a [StockConcentration]
 */
class MissingStockConcentrationException(reactant: Reactant) : Exception() {

    private val reactantName: String

    init {
        this.reactantName = reactant.javaClass.simpleName
    }

    override fun getLocalizedMessage(): String {
        return reactantName + " reactant is missing a stock concentration"
    }
}